tests/openvino/native/quantization/test_weights_compression.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import openvino.runtime as ov
+import pytest
+
+from nncf.quantization import compress_weights
+from tests.openvino.native.models import IntegerModel
+from tests.openvino.native.models import WeightsModel
+
+TEST_MODELS = {
+    IntegerModel: ["gather_2_data", "matmul_1_data", "matmul_2_data"],
+    WeightsModel: ["weights_0", "weights_1"],
+}
+
+
+@pytest.mark.parametrize("model_creator_func", TEST_MODELS)
+def test_compress_weights(model_creator_func):
+    ref_compressed_weights = TEST_MODELS[model_creator_func]
+    model = model_creator_func().ov_model
+    compressed_model = compress_weights(model)
+
+    n_compressed_weights = 0
+    for op in compressed_model.get_ops():
+        if op.get_type_name() == "Constant" and op.get_friendly_name() in ref_compressed_weights:
+            assert op.get_element_type() == ov.Type(np.int8)
+            n_compressed_weights += 1
```

Comment:
I'm afraid this kind of check is not enough to verify that quantization is done correctly.
Could you add a very basic model with ~2-10 weights and calculate expected output?
Or you could catch only quantized weights and compare with the reference. 
The current state of the branch passes the graph test, but accuracy is around 0 on the real model.

Reply:
Sorry, I incorrectly verified accuracy. Seems like there's no issue with accuracy on the current branch. But still, I'd think about better verification than graph test.

Reply:
Added `test_compare_compressed_weights`

Reply:
Thanks!